examples/ultrahdr_app.cpp

```diff
@@ -1240,10 +1235,18 @@ static void usage(const char* name) {
       "    -t    10 bit input transfer function, optional. [0:linear, 1:hlg (default), 2:pq] \n");
   fprintf(stderr,
           "    -q    quality factor to be used while encoding 8 bit image, optional. [0-100], 95 : "
-          "default.\n"
-          "          gain map image does not use this quality factor. \n"
-          "          for now gain map image quality factor is not configurable. \n");
+          "default.\n");
   fprintf(stderr, "    -e    compute psnr, optional. [0:no (default), 1:yes] \n");
+  fprintf(stderr,
+          "    -R    color range for the HDR YUV input, optional. [0:narrow-range (default), "
+          "1:full-range]. \n");
+  fprintf(stderr, "    -s    gain map scale factor, optional. [1, 2, 3, 4 (default)]. \n");
```

Comment:
Any integer should be OK, 4 is default

Reply:
may be any integer is not ok, because values <= 0 are not ok. Values more than 4 can cause gain map image width and height to become zero. Because kMinWidth and kMinHeight are configured to 4 x 4. There no checks in jpegr.cpp after gainmap dimensions are computed in generateGainMap.

Reply:
OK I see.

I think we can leave the low-boundary to 1, and leave the up-boundary open, and pre-check / print an error message if it's too large and triggered the gain map width / height to becoming 0.

Reply:
Done

Reply:
Or perhaps force changing it the the largest-possible value when that happens

Reply:
If gainmap dimensions becomes zero i am choosing min (width, height) as scale factor.

Reply:
Thank you for making this change! One minor comment that I'm feeling if this happens one pixel for the gainmap is too small to contain any useful information. What about using `min(width, height) / DCTSIZE` so that if we see extreme large scaling factor from users input we can maintain the gainmap dimension to at least 16.

Reply:
I have made the change dichen. As we are updating the input configuration with out feedback to the client, is it better if we update it to a value that yields the best image quality (scale factor = 1). Because if the client has configured it accidentally then without feeback he has to debug the cause of any quality issue due to small gainmap image. Also DCTSIZE is jpeg specific and if heir.cpp is introduced this can change to 64 and for avir.cpp this can change to 128. Defaulting to 1, we can update ultrahdr_api.h as well with documentation. Further if we return error if map_width or map_height is zero, then it circumvents all these thoughts.